src/extension.ts

```diff
@@ -129,6 +129,11 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
       if (textEditor?.document === undefined) {
         return;
       }
+
+      if (!isTerraformFile(textEditor?.document)) {
```

Comment:
```suggestion
      if (!isTerraformFile(textEditor.document)) {
```

We've checked in line 129 that `textEditor` and `textEditor.document` is defined, so it's safe to access it without the `?` here.

Reply:
Sorry, I did not mean to suggest removing the check above here. I just meant to remove the `?` in this call (like the proposed change).

We still need to make sure that `textEditor.document` isn't `undefined`, so we can access and pass `uri` to `updateTerraformStatusBar`.

So what I'm suggesting now is: keeping the `if` above as before, removing the `?` (as proposed) and undo the changes to `updateTerraformStatusBar`.

Reply:
Reworked as suggested